docs/operator-manual/config-management-plugins.md

```diff
@@ -201,18 +207,172 @@ volumes:
+  kind: ClusterRole
+  name: argocd-repo-server-getsvcs
+subjects:
+- kind: ServiceAccount
+  name: argocd-repo-server
+  namespace: argocd
+```
+
+To install a plugin as a service, deploy something running your code. This would usually be a deployment, but Argo CD doesn't
+actually care what is behind the service. You can use either an off-the-shelf or custom-built plugin image as sidecar image. This
+should follow the same rules as for a sidecar.
+
+To advertise the service to Argo CD you must provide a service with the label `argocd.argoproj.io/plugin: true`. Each port in the
+service will be treated as a separate plugin, and the name of the port is the name of the plugin, which will be used by applications
+which specify their plugin by name.
+
+This might look like:
+
+```yaml
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: myplugin
+  namespace: argocd
+  labels:
+    argocd.argoproj.io/plugin: "true"
+spec:
+  ports:
+  - name: myplugin
+    port: 8080
+    protocol: TCP
+    targetPort: cmp
+  selector:
+    k8s-app: myplugin
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    k8s-app: myplugin
+  name: myplugin
+  namespace: argocd
+spec:
+  selector:
+    matchLabels:
+      k8s-app: myplugin
+  template:
+    metadata:
+      labels:
+        k8s-app: myplugin
+    spec:
+      containers:
+      - name: myplugin
+        image: myrepo/myplugin:latest
+        imagePullPolicy: Always
+        ports:
+        - containerPort: 8080
+          name: cmp
+          protocol: TCP
+```
+
+!!! important "Double-check these items"
+    1. Make sure to use `argocd-cmp-server` as an entrypoint from the same version of Argo CD. The `argocd-cmp-server` is a lightweight GRPC service that allows Argo CD to interact with the plugin.
+    2. Make sure that sidecar container is running as user 999.
+    3. Make sure that plugin configuration file is present at `/home/argocd/cmp-server/config/plugin.yaml`. It can either be volume mapped via configmap or baked into image.
+	4. The **service** has the label `argocd.argoproj.io/plugin: "true"`
+	5. If you are calling the plugin by name that the **port** name is what you use to call it.
+
+If you'd like to disable this feature do not provide a service account token.
+
+#### Choosing between sidecar and service plugins
+
+If you are debugging your plugin, choose a service to start with. You can then restart the deployments behind the service without
+needing to restart Argo CD. You could also easily develop in an in-cluster development environment this way. If you want to develop
+locally and your cluster can reach your local machine you could even develop using an ExternalName service.
+
+For large mono-repos sidecars will continue to work where services will take longer to transfer the state to the plugin. Argo CD
+doesn't condone large repos anyway. Because the network traffic between the repo-server and the plugin is over a unix socket it is by
```

Comment:
```suggestion
doesn't condone large repos in any way. Because the network traffic between the repo-server and the plugin is over a unix socket it is by
```

Reply:
I've just removed the words instead.